android/app/src/main/java/com/breez/client/plugins/breez/BreezBackup.java

```diff
@@ -0,0 +1,362 @@
+        task.addOnSuccessListener(
+            googleSignInAccount -> {
+                Log.i(TAG, "Sign in success");
+                m_driveClient = Drive.getDriveClient(m_activity, googleSignInAccount);
+                m_driveResourceClient =
+                        Drive.getDriveResourceClient(m_activity, googleSignInAccount);
+
+            })
+            .addOnFailureListener(
+                e -> {
+                    Log.w(TAG, "Sign in failed", e);
+                });
+    }
+
+    @Override
+    public boolean onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == AUTHORIZE_ACTIVITY_REQUEST_CODE) {
+            if (resultCode == RESULT_OK) {
+                getDriveClients(m_googleSignInClient.silentSignIn());
+            }
+            return true;
+        }
+        return false;
+    }
+
+    private void promptToAuthorize() {
+        Intent signInIntent = m_googleSignInClient.getSignInIntent();
+        m_activity.startActivityForResult(signInIntent, AUTHORIZE_ACTIVITY_REQUEST_CODE);
+    }
+
+    @Override
+    public void onMethodCall(MethodCall call, MethodChannel.Result result) {
+        if (call.method.equals("authorize")) {
+            if (m_driveResourceClient == null) {
+                promptToAuthorize();
+            }
+            else {
+                m_result.success(true);
+            }
+        }
+        if (call.method.equals("backup")) {
+            List<String> paths = call.argument("paths");
+            String nodeId = call.argument("nodeId").toString();
+            createNodeIdFolder(paths, nodeId);
+        }
+        if (call.method.equals("listBackupFolders")) {
+            listBackupFolders();
+        }
+        if (call.method.equals("restore")) {
+            if (m_driveResourceClient == null) {
+                promptToAuthorize();
+            }
+            else {
+                m_result = result;
+                String nodeId = call.argument("nodeId").toString();
+                if (nodeId.isEmpty()) {
+                    listBackupFolders();
+                } else {
+                    getNodeIdFolder(nodeId);
+                }
+            }
+        }
+    }
+
+    private void notifyBackupDisabled() {
+        m_eventsListener.success(false);
+    }
+
+    private void notifyBackupEnabled() {
+        m_eventsListener.success(true);
+    }
+
+    private void updateBackupFiles(List<String> paths, DriveFolder nodeIdFolder) {
+        for (String path: paths) {
+            m_driveResourceClient.createContents()
+                .continueWithTask(task -> {
+                    DriveContents contents = task.getResult();
+                    OutputStream outputStream = contents.getOutputStream();
+
+                    File file = new File(path);
+                    FileInputStream inputStream = new FileInputStream(file);
+
+                    byte[] buffer = new byte[inputStream.available()];
```

Comment:
available function doesn't return all the bytes in the stream (see here: https://docs.oracle.com/javase/7/docs/api/java/io/FileInputStream.html#available())
Reading from InputStream in java requires a loop until read method returns -1.
Something like this:
```
do {
    byte[] buf = new byte[1024];
     i = inputStream.read(buf);               
} while (i != -1);
```